### ISSUE TITLE
let_it_be reload alias not functioning

### ISSUE THREAD

**Issue by samstickland** [palkan/test-prof#144]
**Ruby Version:** 2.5.1

**Framework Version (RSpec, Minitest, FactoryGirl, Rails, whatever):** RSpec 3.7.0

**TestProf Version:** 0.9.0

### What did you do?

```
TestProf::LetItBe.configure do |config|
  # define an alias with `reload: true` by default
  config.alias_to :let_it_be_with_reload, reload: true
end
```

### What did you expect to happen?

I expected that `let_it_be_with_reload(:x)` would function identically to `let_it_be(:x, reload: true)`

### What actually happened?

The created object is not removed at when leaving the the RSpec context.

The following RSpec test illustrates the problem:

```
RSpec.describe 'let_it_be functions correctly', order: :defined do
  let_it_be(:email) { FactoryGirl.build(:email) }

  context 'without reload' do
    context 'first context' do
      let_it_be(:user_one) { FactoryGirl.create :c_user, email: email }

      it 'has only one user' do
        expect(User.count).to eq 1
      end
    end

    context 'second context' do
      let_it_be(:user_one) { FactoryGirl.create :c_user, email: email }

      it 'has only one user' do
        expect(User.count).to eq 1
      end
    end
  end

  context 'with explicit reload' do
    context 'first context' do
      let_it_be(:user_one, reload: true) { FactoryGirl.create :c_user, email: email }

      it 'has only one user' do
        expect(User.count).to eq 1
      end
    end

    context 'second context' do
      let_it_be(:user_one, reload: true) { FactoryGirl.create :c_user, email: email }

      it 'has only one user' do
        expect(User.count).to eq 1
      end
    end
  end

  context 'with reload alias' do
    context 'first context' do
      let_it_be_with_reload(:user_one) { FactoryGirl.create :c_user, email: email }

      it 'has only one user' do
        expect(User.count).to eq 1
      end
    end

    context 'second context' do
      let_it_be_with_reload(:user_one) { FactoryGirl.create :c_user, email: email }

      it 'has only one user' do
        expect(User.count).to eq 1
      end
    end
  end
end
```

`let_it_be` works correctly, it also works correctly with the explicit reload, but it does not work correctly with the alias:

```
$ rspec -fdoc spec/let_it_be_spec.rb

Randomized with seed 55677

let_it_be functions correctly
  without reload
    first context
      has only one user
    second context
      has only one user
  with explicit reload
    first context
      has only one user
    second context
      has only one user
  with reload alias
    first context
      has only one user
    second context
      has only one user (FAILED - 1)
```

**Comment by palkan** [comment]
What is the failure message?

What happens if you runs specs in random order (`--order rand`)?

**Comment by samstickland** [comment]
The failure method is a duplicate database key (as duplicate email addresses are not allowed):

```
  1) let_it_be functions correctly with reload alias second context has only one user
     Failure/Error: FactoryGirl.create factory, attributes

     ActiveRecord::RecordInvalid:
       Validation failed: Email has already been taken
     # ./spec/factories/users.rb:12:in `block (3 levels) in <top (required)>'
     # ./spec/lib/let_it_be_spec.rb:52:in `block (4 levels) in <top (required)>'
     # ./spec/support/let_all.rb:9:in `instance_exec'
     # ./spec/support/let_all.rb:9:in `block in let_it_be_with_reload'
```

If I run it in a random order then any tests that get executed after the first 'with reload alias' test fail. Here are some examples:

```
Randomized with seed 41860

let_it_be functions correctly
  with reload alias
    second context
      has only one user
    first context
      has only one user (FAILED - 1)
  without reload
    first context
      has only one user (FAILED - 2)
    second context
      has only one user (FAILED - 3)
  with explicit reload
    first context
      has only one user (FAILED - 4)
    second context
      has only one user (FAILED - 5)
```

```
let_it_be functions correctly
  without reload
    first context
      has only one user
    second context
      has only one user
  with reload alias
    first context
      has only one user
    second context
      has only one user (FAILED - 1)
  with explicit reload
    second context
      has only one user (FAILED - 2)
    first context
      has only one user (FAILED - 3)
```

```
Randomized with seed 36132

let_it_be functions correctly
  with explicit reload
    first context
      has only one user
    second context
      has only one user
  without reload
    second context
      has only one user
    first context
      has only one user
  with reload alias
    first context
      has only one user
    second context
      has only one user (FAILED - 1)
```

The error message is the same for each failed test:

```
     ActiveRecord::RecordInvalid:
       Validation failed: Email has already been taken
```

**Comment by palkan** [comment]
What's inside the `./spec/support/let_all.rb` file?

**Comment by samstickland** [comment]
Oh, I'd missed that that was getting picked up! We had a previous implementation of a custom method which we called `let_all!` (that was built on the rspec_nested_transactions gem) that does the same thing as `let_it_be(..., reload: true)`

I did a search and replace for `let_all!` to `let_it_be_with_reload` and it also renamed the `let_all!` method definition to `let_it_be_with_reload` and override the alias. Doh!

I've deleted the `spec/support/let_all.rb` file and it's resolved the problem.

Thanks for the spot, sorry about the waste of your time!
